### PR TITLE
New PR templates to help with new process

### LINKED ISSUE #322: Special PR template for staging -> master PRs
As we are discussing some of the nitty gritty for how staging -> master PRs should work, I think its clear that these PRs are quite a different beast than what our current PR template prescribes. 

As we determine what kinds of checks and considerations need to be made for a staging -> master PR, we should add that to this new PR template. 

Some things I think should be included as of now: 
```
What overall changes are being included in this "make it live" PR? 
What staging commits/PR #'s are included in this "make it live" PR?
Did we do a last check with the affected html files using html preview? 
```

### LINKED ISSUE #331: Make a PR template for "hotfixes"
### Background

In the discussion about the PR process in #313, [@jashpiro noted](https://github.com/AlexsLemonade/refinebio-examples/pull/313#commitcomment-43518311) we should make sure that hotfix PRs, which should involve two PRs, one to `master` and secondly one to `staging` should note the `master` PR so we know they've been reviewed and the process can be expedited. @jaclyn-taroni mentioned this would be a good place for a separate PR template (separate from #322). 

### Problem

hotfix PRs will involve the same change made to two branches: staging and master. We don't want this to increase reviewer burden aka in-depth review of the same essential change twice. 

### What are the recommended next steps?
The hotfix PR template should remind the person who files it to mention if this change has been reviewed on another PR and link to it. This was the second PR to `staging` can go by more quickly. 

I don't think this means we need two PR templates for hotfixes, just one template that has a prompt that encourages the user to mention a previous PR if applicable. 

This template should also include other basics like "strategy" and "problem" and perhaps other items I'm not aware of at this time but we might figure out when we try to do our first "hotfix".

### PR DESCRIPTION
**Pull request by cansavvy**
## Purpose:
Closes #331 and closes #322 

With the new features and complications of our PR process, we could use some new and more specific PR templates to help guide us through the process. 

## Strategy
I followed this blog post and took a look at the github docs it links to: https://github.blog/2018-01-25-multiple-issue-and-pull-request-templates/
I am not sure how to know if this will for sure work before we merge it. Those instructions say something about changing the url but it's unclear where they are referring to?

As I was adding a PR template for hotfixes and "make it live" PRs, I realized we also could use an "Other" so now we have 4 PR templates:
- hotfix-pr.md
- <del> make-it-live-pr.md </del> publish-pr.md
- new-analysis-pr.md
- other-issue.md

 Is this too many to be helpful? 
## Concerns/Questions for reviewers:
- Is there any info the new PR templates should probably have prompts for but currently don't? 
- Are the prompts sensible? 
- Is this too many PR templates?

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/hotfix-pr.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+name: Hotfix PR
```
Comment (cansavvy) [inline .github/PULL_REQUEST_TEMPLATE/hotfix-pr.md:2]: Does this also work in PR templates? I am not sure.

File: .github/PULL_REQUEST_TEMPLATE/new-analysis-pr.md [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
+
+---
+
+## Analysis Purpose
+
+<!-- What new analysis issue(s) does your PR address? -->
+
+## Pull Request Stage
+
+<!-- Delete one of these to let the reviewer know what stage this analysis example is at -->
```
Comment (cbethell) [inline .github/PULL_REQUEST_TEMPLATE/new-analysis-pr.md:16]:
```suggestion
<!-- Select one of these and delete the other to let the reviewer know what stage this analysis example is at -->
```

If we use the wording above, we would want to propagate it here.

File: .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+---
+
+## Changes Being Made Live
+
+<!-- What overall changes are being included in this "make it live" PR and are they all ready to be made live? -->
+
+## List of Commits/PR's Included
+
+<!-- List the staging commits/PR #'s that are included PR? with a general and quick summary of what each does -->
```
Comment (cbethell) [inline .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md:16]:
```suggestion
<!-- List the staging commits/PR #'s that are included with a general and quick summary of what each does -- start by finishing the prompt below -->
```
To make it clear that below prompt would need to be modified to be relevant to the PR, what do you think?

File: .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+- PR# Added such and such
+
+## Link to html Preview
+
+For reviewing purposes, you can take a browse through htmlpreview: http://htmlpreview.github.io/?https://github.com/AlexsLemonade/refinebio-examples/gh-pages-stages/01-getting-started/getting-started.html
+
+## Make it Live Checklist:
```
Comment (cbethell) [inline .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md:24]:
```suggestion
## Publish Checklist:
```

Propagating the "make-it-live" -> "publish" wording change

File: .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+- [ ] Does everything look good using html preview -- having taken a particularly close look at any new html files?
+
+- [ ] Have any new examples added to the [refinebio-examples feedback survey](https://app.hubspot.com/forms/5187852/a50f293c-1ef4-4ee1-b7ee-c563afe2ad5c/performance)? Ideally, don't click publish until this PR is merged (changes are autosaved).
```
Comment (cbethell) [inline .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md:28]:
```suggestion
- [ ] Have any new examples been added to the [refinebio-examples feedback survey](https://app.hubspot.com/forms/5187852/a50f293c-1ef4-4ee1-b7ee-c563afe2ad5c/performance)? Ideally, don't click publish until this PR is merged (changes are autosaved).
```

**Review by cbethell — COMMENTED** [review]
Left a few suggestions below but overall this PR looks good.

Re your "Is this too many PR templates to be helpful?" question, I gather that we would use the templates in this PR like the following

- hotfix-pr.md - there may be a package citation that was left out when the analysis was merged to master so we would want to get that in as quickly as possible with a hotfix PR
- make-it-live-pr.md - after some changes have been merged to `staging` (for a new analysis, existing analysis, etc.), we will want to publish them to `master` and we will want to use this PR template for said PR
- new-analysis-pr.md - we would want to use this template when filing a PR that adds a new analysis to the repo, which means this PR should contain _at least_ two new files (a new `.Rmd` and a new `.html` file)
- other-issue.md - we would use this template for any issue not pertaining to a new analysis or a hotfix i.e. an update to an existing analysis that may not be so straightforward or an addition to documentation or a `README.md` file

If my above understanding of the use cases for each template is correct, I would say that that are not too numerous to be helpful as they each have a particular and necessary use case.

If my above understanding is incorrect, on the other hand, we may want to revisit the use case that we would want each template to address.

File: .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
+---
+name: Make it live PR
```
[the remark this answers is not in the document]
Comment (cansavvy) [reply .github/PULL_REQUEST_TEMPLATE/make-it-live-pr.md:2]: Yes. I realized I missed this one.

**Comment by cansavvy** [comment]
I agree with your summary above, @cbethell and if you think this set up will be helpful, then we can try it out! (and revisit as needed). 

I think I've incorporated your changes, let me know what you think.

**Review by cbethell — APPROVED** [review]
LGTM!

**Comment by cansavvy** [comment]
I think it makes sense to merge these so we can start to try them out. As we find things we want tweaked, we can file issues and revisit! Aka if we think they are reasonable now, let's try it out and keep in mind these are not written in stone.

**Pull request merged**